src/Ng/Json/InterceptingJsonReader.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+
+namespace Ng.Json
+{
+    public abstract class InterceptingJsonReader
+        : JsonReader
+    {
+        private static readonly ConcurrentDictionary<string, Regex> CachedRegexes = new ConcurrentDictionary<string, Regex>();
+
+        protected JsonReader InnerReader { get; }
+        protected List<Regex> PropertiesToIntercept { get; }
+
+        protected InterceptingJsonReader(JsonReader innerReader, IEnumerable<string> interceptPaths)
+        {
+            InnerReader = innerReader;
+            InnerReader.DateParseHandling = DateParseHandling.None;
+
+            PropertiesToIntercept = interceptPaths
+                .Distinct(StringComparer.Ordinal)
+                .Select(path => MakeRegex(path))
+                .ToList();
+        }
+
+        private static Regex MakeRegex(string jsonPath)
+        {
+            Regex expression = null;
+            if (!CachedRegexes.TryGetValue(jsonPath, out expression))
+            {
+                var pattern = jsonPath
+                    .Replace("[*]", @"\[\d+\]")
+                    .Replace(".", @"\.");
+
+                expression = new Regex("^" + pattern);
```

Comment:
Just discussed this with Joel - and he points out that we should have closed this one down with a $ i.e. something like:
     expression = new Regex("^" + pattern + "$");

(Though we haven't tested that one.)

The reason we think this hasn't mattered is that the paths we happen to have all terminating - precisely to the property in question. In other words we think this works as is but is a little fragile.